### PR TITLE
Collectives 'features' endpoint

### PR DESCRIPTION
**Pull request by Betree**
:point_right: Please read https://github.com/opencollective/opencollective/issues/2807 for the motivations and specs of this feature.

A proposal for a new `features` endpoint for Collectives, currently on API V1 only but meant to be copied for API V2. Some features are not entirely covered yet (ie. we don't check the data for conversations) because I'd like to get some feedback from the team before going further.

### CODE REVIEW

File: server/graphql/common/features.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  promise: Promise<number | boolean>,
+  fallback = FEATURE_STATUS.AVAILABLE,
+): Promise<FEATURE_STATUS> => {
+  return promise.then(result => (result ? FEATURE_STATUS.ACTIVE : fallback));
+};
+
+/**
+ * Returns a resolved that will give the `FEATURE_STATUS` for the given collective/feature.
+ */
+export const getFeatureStatusResolver = (feature: FEATURE) => async (collective): Promise<FEATURE_STATUS> => {
```
Comment (kewitz) [inline server/graphql/common/features.ts:16]: :100:

**Comment by Betree** [comment]
To add (for conditionally displaying the sections on the collective page):
- `CONTRIBUTE` (always allowed, only active if member of another collective - Host, Backer or Attendee)
- `MAKE_RECURRING_CONTRIBUTIONS` (only active if there are recurring contributions)

**Comment by Betree** [comment]
Note: this should be added on GQLV2 as well for feature parity

**Comment by Betree** [comment]
Closing in favor of https://github.com/opencollective/opencollective-api/pull/3992

**Pull request closed without merging**